testsuite/tests/language-extensions/language_extensions.ml

```diff
@@ -2,118 +2,148 @@
    include ocamlcommon
    flags = "-I ${ocamlsrcdir}/parsing"
    reference = "${test_source_directory}/reference.txt"
-   * skip
-   reason = "local_ currently always parses"
 *)
 
+(* Currently, this test case only works with modular extensions, as they fail
+   during parsing and we test via parsing. *)
+
```

Comment:
Skimming through, I don't think it's quite worth a careful review of these tests. It looks broadly that they're working and that we will discover if they're not.